homeassistant/helpers/entity.py

```diff
@@ -315,6 +317,15 @@ def unique_id(self) -> str | None:
         """Return a unique ID."""
         return self._attr_unique_id
 
+    @property
+    def modern_name(self) -> bool:
```

Comment:
`entity_name` is better, I think. Modern is temporary.

Reply:
I think it should be temporary. At some point, we should be able to remove this. As the `name` property is preserved, even if an acient integration/entity is still using the old style; it would not break hard.

Reply:
Renamed from `modern_name` to `has_entity_name`